projects/bootloader/src/flash_application_code.c

```diff
@@ -0,0 +1,73 @@
+#include "flash_application_code.h"
+
+#include "bootloader_can.h"
+#include "bootloader_datagram_defs.h"
+#include "can_datagram.h"
+#include "crc32.h"
+#include "dispatcher.h"
+#include "flash.h"
+#include "flash_application_code.pb.h"
+#include "pb_decode.h"
+#include "stdint.h"
+
+// process:
+// 1. listen for datagram id 8
+// 2. respond
+// 3. listen for datagram id 9
+// 4. flash
+// 5. respond (waits for flash to finish)
+// 5. repeat 3 - 5
+
+#define APPLICATION_START_PAGE 2  // start of the application page in flash
```

Comment:
The application's flash actually starts at the address `BOOTLOADER_APPLICATION_START` from bootloader_mcu.h.